src/sentry/static/sentry/app/components/dataExport.tsx

```diff
@@ -46,18 +46,22 @@ class DataExport extends React.Component<Props, State> {
       payload: {queryType, queryInfo},
     } = this.props;
     try {
-      const {id: dataExportId} = await api.requestPromise(
+      const [data, _, response] = await api.requestPromise(
         `/organizations/${slug}/data-export/`,
         {
+          includeAllArgs: true,
           method: 'POST',
           data: {
             query_type: queryType,
             query_info: queryInfo,
           },
         }
       );
+      const {id: dataExportId} = data;
       addSuccessMessage(
-        t("Sit tight. We'll shoot you an email when your data is ready for download.")
+        response?.status === 201
+          ? t("Sit tight. We'll shoot you an email when your data is ready for download.")
+          : t("It looks like we're already working on it. Sit tight, we'll email you.")
```

Comment:
This is Point 1. Show a different toast message when the user refreshes the page and tries the same download. (There is a ticket to eventually save the state across refreshes, but it proves a little more difficult, so this is a temp fix)